scipy/stats/_hypotests.py

```diff
@@ -389,3 +390,192 @@ def _get_wilcoxon_distr(n):
                          "statistic is not implemented for n={}".format(n))
 
     return np.array(cnt, dtype=int)
+
+
+def _pval_cvm_2samp_exact(s, nx, ny):
+    """
+    Compute the exact p-value of the Cramer-von Mises two-sample test
+    for a given value s (float) of the test statistic by enumerating
+    all possible combinations. nx and ny are the sizes of the samples.    
+    """
+    z = np.arange(1, nx+ny+1)
+    rangex = np.arange(1, nx+1)
+    rangey = np.arange(1, ny+1)
+
+    us = []
+    # for the first sample x, select all possible combinations of
+    # the ranks of the elements of x in the combined sample
+    # note that the acutal values of the samples are irrelevant
+    # once the ranks of the elements of x are fixed, the
+    # ranks of the elements in the second sample
+    # that z = 1, ..., nx+ny contains all possible ranks for a loop over all 
+    for c in combinations(z, nx):
+        x = np.array(c)
+        # the ranks of the second sample y are given by the set z - x
+        # note than one needs to shift by -1 since the lowest rank is 1
+        mask = np.ones(nx+ny, bool)
+        mask[x-1] = False
+        y = z[mask]
```

Comment:
Would you consider breaking this part out into a generator? I would like to use it elsewhere.